src/Module.cpp

```diff
@@ -266,8 +275,27 @@ Module Module::resolve_submodules() const {
     for (const auto &buf : buffers()) {
         lowered_module.append(buf);
     }
+    for (const auto &ec : external_code()) {
+        lowered_module.append(ec);
+    }
     for (const auto &m : submodules()) {
         Module copy(m.resolve_submodules());
```

Comment:
We make a copy with submodules resolved, and laboriously add the external code blocks to it... but then don't do anything with the copy (we compile "m", not "copy"). Is this right? If so, it needs a comment about what it accomplishes.